pkg/kmm/preflightvalidationocp_test.go

```diff
@@ -0,0 +1,328 @@
+		},
+		{
+			name:              "",
+			namespace:         defaultPreflightNamespace,
+			expectedPreflight: nil,
+			expectedErr:       "PreflightValidationOCP 'name' cannot be empty",
+		},
+	}
+
+	for _, testCase := range testCases {
+		testSettings := clients.GetTestClients(clients.TestClientParams{})
+
+		testBuilder := NewPreflightValidationOCPBuilder(testSettings, testCase.name, testCase.namespace)
+
+		if testCase.expectedErr == "" {
+			assert.NotNil(t, testBuilder)
+			assert.Equal(t, testCase.expectedPreflight.Name, testBuilder.Definition.Name)
+			assert.Equal(t, testCase.expectedPreflight.Namespace, testBuilder.Definition.Namespace)
+		} else {
+			assert.Equal(t, testCase.expectedErr, testBuilder.errorMsg)
+		}
+	}
+}
+
+func TestPreflightWithReleaseImage(t *testing.T) {
+	testCases := []struct {
+		image       string
+		expecterErr string
+	}{
+		{
+			image:       "some-image",
+			expecterErr: "",
+		},
+		{
+			image:       "",
+			expecterErr: "invald 'image' argument can not be nil",
+		},
+	}
+
+	for _, testCase := range testCases {
+		testBuilder := buildValidTestPreflight()
+		testBuilder.WithReleaseImage(testCase.image)
+
+		if testCase.expecterErr == "" {
+			assert.Equal(t, testBuilder.Definition.Spec.ReleaseImage, testCase.image)
+		} else {
+			assert.Equal(t, testCase.expecterErr, testBuilder.errorMsg)
+		}
+	}
+}
+
+func TestPreflightWithUseRTKernel(t *testing.T) {
+	testCases := []struct {
+		useRT bool
+	}{
+		{
+			useRT: true,
+		},
+		{
+			useRT: false,
+		},
+	}
+
+	for _, testCase := range testCases {
+		testBuilder := buildValidTestPreflight()
+		testBuilder.WithUseRTKernel(testCase.useRT)
+		assert.Equal(t, testCase.useRT, testBuilder.Definition.Spec.UseRTKernel)
+	}
+}
+
+func TestPreflightWithPushBuiltImage(t *testing.T) {
+	testCases := []struct {
+		push bool
+	}{
+		{
+			push: true,
+		},
+		{
+			push: false,
+		},
+	}
+
+	for _, testCase := range testCases {
+		testBuilder := buildValidTestPreflight()
+		testBuilder.WithPushBuiltImage(testCase.push)
+		assert.Equal(t, testCase.push, testBuilder.Definition.Spec.PushBuiltImage)
+	}
+}
+
+func TestPreflightWithOptions(t *testing.T) {
+	testBuilder := buildValidTestPreflight()
+
+	testBuilder.WithOptions(func(builder *PreflightValidationOCPBuilder) (*PreflightValidationOCPBuilder, error) {
+		return builder, nil
+	})
+
+	assert.Equal(t, "", testBuilder.errorMsg)
+
+	testBuilder.WithOptions(func(builder *PreflightValidationOCPBuilder) (*PreflightValidationOCPBuilder, error) {
+		return builder, errors.New("error")
+	})
+
+	assert.Equal(t, "error", testBuilder.errorMsg)
+}
+
+func TestPullPreflightValidationOCP(t *testing.T) {
+	testCases := []struct {
+		name                string
+		namespace           string
+		expectedError       bool
+		addToRuntimeObjects bool
+		expectedErrorText   string
+	}{
+		{
+			name:                "test",
+			namespace:           "testns",
+			expectedError:       false,
+			addToRuntimeObjects: true,
```

Comment:
expectedErrorText  should be explicitly set as well.

Reply:
Fixed ! Thanks !